Library/Formula/llvm.rb

```diff
@@ -116,18 +152,39 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    # compiler-rt has some iOS simulator features that require i386 symbols
+    # I'm assuming the rest of clang needs support too for 32-bit compilation
+    # to work correctly, but if not, perhaps universal binaries could be
+    # limited to compiler-rt. llvm makes this somewhat easier because compiler-rt
+    # can almost be treated as an entirely different build from llvm.
+    ENV.permit_arch_flags
+
+    prep = lambda do |resource_name, path, required|
+      if build.with?(resource_name) &&
+         (required.nil? || build.with?(required))
+        (buildpath/path).install resource(resource_name)
+      elsif build.with?(resource_name) && build.without?(required)
+        raise resource_name + " requires " + required + " to build."
+      end
```

Comment:
I have no idea what's going on here. Please just inline these instead of using a lambda.